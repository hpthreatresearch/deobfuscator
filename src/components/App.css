* {
  --white: #fcfdfd;
  --black-50: #f3f5f7;
  --black-100: #e7e7e8;
  --black-200: rgb(196, 196, 197);
  --black-300: rgb(160, 160, 162);
  --black-400: rgb(89, 89, 92);
  --black-500: #121216;
  --black-600: #101014;
  --black-700: #0b0b0d;
  --black-800: #08080a;
  --black-900: #050507;
}

*::-webkit-scrollbar {
  opacity: 0.75;
  width: 10px;
}
*::-webkit-scrollbar-track {
  background: rgb(0, 0, 0, 0);
}
*::-webkit-scrollbar-thumb {
  background: var(--black-200);
  border-radius: 5px;
}

html {
  color: var(--black);
}

h3 {
  font-size: 1.5rem;
  margin-bottom: 0;
}
.background-gray {
  background-color: var(--black-50);
}

.background-white {
  background-color: var(--white);
}

.bake {
  margin-top: auto;
}

.CodeMirror-wrap pre {
  word-break: break-word;
}

.spawn-card {
  background-color: #bce8f1;
  color: #31647e;
  border: 1px solid #9fd6e1;
  padding: 12px;
  cursor: grab;
  display: flex;
  align-items: center;
}

.spawn-card > .icon {
  margin-left: auto;
  cursor: pointer;
}

.spawn-card ~ .spawn-card {
  border-top-width: 0px;
}

.border-white {
  border: 2px solid var(--white);
}

.next-module {
  border: 2px solid red;
}

.module {
  border: 1px solid #38512d;
  padding: 12px 25px;
  background-color: #dff0d8;
  color: #38512d;
  cursor: move;
}

.loop-container {
  position: relative;
  border: 1px solid #38512d;
  border-right: none;

  background-color: var(--white);
  margin-right: -26px;
  margin-left: 5px;
  margin-bottom: 18px;
  margin-top: 18px;

  padding-top: 10px;
  padding-left: 10px;
  padding-bottom: 10px;
}

.next-module > * > .loop-container::before {
  content: "";
  border: solid 2px red;
  border-right: none;

  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: -2px;
  pointer-events: none;
}

.loop-container::after {
  content: "";
  background-color: var(--white);
  width: 2px;

  height: 100%;
  position: absolute;
  top: 0;
  right: -2px;
  pointer-events: none;
}

.fill-under-title {
  height: calc(100% - 50px);
  max-height: calc(100% - 50px);
}

.Í¼2 .cm-gutters {
  background-color: var(--white) !important;
  border: none !important;
}

.overflow-none-100 {
  height: 100%;
  max-height: 100%;
  overflow: hidden;
}

.modal-content {
  background-color: var(--white) !important;
}

blockquote {
  margin: 0;
  font-style: italic;
}

blockquote p {
  padding: 15px;
  background: var(--black-100);
  border-radius: 5px;
}

.cursor-pointer {
  cursor: pointer;
}

.output {
  font-family: monospace;
  line-height: 1.4;
}

.output > pre {
  padding-bottom: 15px;
  border-bottom: 1px solid var(--black-100);
  margin-bottom: 15px;
  margin-left: 15px;
  margin-right: 15px;
}

.output > pre,
.output > * > pre {
  font-size: unset;
}

.my-progress-bar {
  position: absolute;
  top: 0;
  z-index: 10;
  left: 0;
  width: 100%;
  border-radius: 0 !important;
  height: 8px;
}