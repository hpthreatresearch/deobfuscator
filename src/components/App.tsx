// Â© Copyright 2023 HP Development Company, L.P.
import type { ReactSetState } from "@core/types";
import { useCallback, useState } from "react";
import "./App.css";
import { ObfuscatorPanel } from "./ObfuscatorPanel";
import { MainContainer } from "./basic/MainContainer";
import { SpawnArea } from "./dragging/SpawnArea/SpawArea";
import { Recipe } from "./recipe-builder/recipe/Recipe/Recipe";
import { Sandbox } from "./sandbox";

function App() {
  const loadingState = useState<boolean>(false);

  const [code, setCode] = useState("");

  const [obfCode, setObfCode] = useState("");

  const [mappings, setMappings] = useState<number[][]>([]);

  const [isOutDated, setIsOutDated] = useState(false);

  const [loading] = loadingState;

  /**
   * When recipe sets mappings, the mappings will no longer be outdated
   */
  const _setMappings: ReactSetState<number[][]> = useCallback((...args) => {
    setIsOutDated(false);
    return setMappings(...args);
  }, []);

  return (
    <MainContainer>
      <SpawnArea />
      <Recipe
        obfCode={obfCode}
        setCode={setCode}
        setMappings={_setMappings}
        loadingState={loadingState}
      />
      <ObfuscatorPanel
        codeState={[code, setCode]}
        mappingsState={[mappings, setMappings]}
        loading={loading}
        setObfCode={setObfCode}
        isOutdatedState={[isOutDated, setIsOutDated]}
      />
      <Sandbox />
    </MainContainer>
  );
}

export default App;
