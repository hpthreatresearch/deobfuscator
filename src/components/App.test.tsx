// Â© Copyright 2023 HP Development Company, L.P.
import renderer from "react-test-renderer";
import App from "./App";

describe(App, () => {
  it("", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
