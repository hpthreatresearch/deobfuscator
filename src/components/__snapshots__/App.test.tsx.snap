// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  1`] = `
<div
  className="overflow-none-100 p-1 mb-2 background-gray container-fluid"
>
  <div
    className="overflow-none-100 gx-1 row"
  >
    <div
      className="d-flex flex-column overflow-none-100 col-xl-2"
    >
      <div
        className="title-block"
      >
        <h3>
          Modules
        </h3>
        <div
          className="d-flex flex-row"
        />
      </div>
      <div
        className="p-2 mb-2 h-100 mh-100 overflow-auto"
      >
        <input
          className="mb-3 form-control"
          onChange={[Function]}
          placeholder="Search..."
          value=""
        />
        <div
          className="module-library"
        >
          <div
            className="module-library-title "
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-arrows-up-down me-3"
              data-icon="arrows-up-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M182.6 9.4c-12.5-12.5-32.8-12.5-45.3 0l-96 96c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L128 109.3V402.7L86.6 361.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l96 96c12.5 12.5 32.8 12.5 45.3 0l96-96c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 402.7V109.3l41.4 41.4c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-96-96z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            Control flow
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-down chevron"
              data-icon="chevron-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
        </div>
        <div
          className="module-library"
        >
          <div
            className="module-library-title "
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-clone me-3"
              data-icon="clone"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M288 448H64V224h64V160H64c-35.3 0-64 28.7-64 64V448c0 35.3 28.7 64 64 64H288c35.3 0 64-28.7 64-64V384H288v64zm-64-96H448c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64H224c-35.3 0-64 28.7-64 64V288c0 35.3 28.7 64 64 64z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            Remove proxies
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-down chevron"
              data-icon="chevron-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
        </div>
        <div
          className="module-library"
        >
          <div
            className="module-library-title "
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-book-open me-3"
              data-icon="book-open"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M249.6 471.5c10.8 3.8 22.4-4.1 22.4-15.5V78.6c0-4.2-1.6-8.4-5-11C247.4 52 202.4 32 144 32C93.5 32 46.3 45.3 18.1 56.1C6.8 60.5 0 71.7 0 83.8V454.1c0 11.9 12.8 20.2 24.1 16.5C55.6 460.1 105.5 448 144 448c33.9 0 79 14 105.6 23.5zm76.8 0C353 462 398.1 448 432 448c38.5 0 88.4 12.1 119.9 22.6c11.3 3.8 24.1-4.6 24.1-16.5V83.8c0-12.1-6.8-23.3-18.1-27.6C529.7 45.3 482.5 32 432 32c-58.4 0-103.4 20-123 35.6c-3.3 2.6-5 6.8-5 11V456c0 11.4 11.7 19.3 22.4 15.5z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            Improve readability
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-down chevron"
              data-icon="chevron-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
        </div>
        <div
          className="module-library"
        >
          <div
            className="module-library-title "
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-filter me-3"
              data-icon="filter"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.9 54.9C10.5 40.9 24.5 32 40 32H472c15.5 0 29.5 8.9 36.1 22.9s4.6 30.5-5.2 42.5L320 320.9V448c0 12.1-6.8 23.2-17.7 28.6s-23.8 4.3-33.5-3l-64-48c-8.1-6-12.8-15.5-12.8-25.6V320.9L9 97.3C-.7 85.4-2.8 68.8 3.9 54.9z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            Simplify code
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-down chevron"
              data-icon="chevron-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
        </div>
        <div
          className="module-library"
        >
          <div
            className="module-library-title "
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-bullseye me-3"
              data-icon="bullseye"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 256A192 192 0 1 0 64 256a192 192 0 1 0 384 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256 80a80 80 0 1 0 0-160 80 80 0 1 0 0 160zm0-224a144 144 0 1 1 0 288 144 144 0 1 1 0-288zM224 256a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            Target tool
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-down chevron"
              data-icon="chevron-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
        </div>
        <div
          className="module-library"
        >
          <div
            className="module-library-title "
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-crop-simple me-3"
              data-icon="crop-simple"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128 32c0-17.7-14.3-32-32-32S64 14.3 64 32V64H32C14.3 64 0 78.3 0 96s14.3 32 32 32H64V384c0 35.3 28.7 64 64 64H352V384H128V32zM384 480c0 17.7 14.3 32 32 32s32-14.3 32-32V448h32c17.7 0 32-14.3 32-32s-14.3-32-32-32H448l0-256c0-35.3-28.7-64-64-64L160 64v64l224 0 0 352z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            Trim
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-down chevron"
              data-icon="chevron-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="h-100 mh-100 background-white p-0 m-0 col-xl-2"
    >
      <div
        className="title-block"
      >
        <h3>
          Recipe
        </h3>
        <div
          className="d-flex flex-row"
        >
          <div
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-floppy-disk "
              data-icon="floppy-disk"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V173.3c0-17-6.7-33.3-18.7-45.3L352 50.7C340 38.7 323.7 32 306.7 32H64zm0 96c0-17.7 14.3-32 32-32H288c17.7 0 32 14.3 32 32v64c0 17.7-14.3 32-32 32H96c-17.7 0-32-14.3-32-32V128zM224 288a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
          <div
            aria-label="load-recipe"
            data-testid="load-recipe"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-folder-open "
              data-icon="folder-open"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M88.7 223.8L0 375.8V96C0 60.7 28.7 32 64 32H181.5c17 0 33.3 6.7 45.3 18.7l26.5 26.5c12 12 28.3 18.7 45.3 18.7H416c35.3 0 64 28.7 64 64v32H144c-22.8 0-43.8 12.1-55.3 31.8zm27.6 16.1C122.1 230 132.6 224 144 224H544c11.5 0 22 6.1 27.7 16.1s5.7 22.2-.1 32.1l-112 192C453.9 474 443.4 480 432 480H32c-11.5 0-22-6.1-27.7-16.1s-5.7-22.2 .1-32.1l112-192z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
          <div
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-trash "
              data-icon="trash"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                fill="currentColor"
                style={{}}
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="drop-area"
      >
        <div
          className="p-2 mb-2 overflow-auto mh-100 h-100"
        />
      </div>
      <div
        className="button-bar"
      >
        <button
          className="mx-3 btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Step
        </button>
        <button
          className="flex-grow-1 mx-3 btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-cake-candles "
            data-icon="cake-candles"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={
              {
                "marginRight": "10px",
              }
            }
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M86.4 5.5L61.8 47.6C58 54.1 56 61.6 56 69.2V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L105.6 5.5C103.6 2.1 100 0 96 0s-7.6 2.1-9.6 5.5zm128 0L189.8 47.6c-3.8 6.5-5.8 14-5.8 21.6V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L233.6 5.5C231.6 2.1 228 0 224 0s-7.6 2.1-9.6 5.5zM317.8 47.6c-3.8 6.5-5.8 14-5.8 21.6V72c0 22.1 17.9 40 40 40s40-17.9 40-40V69.2c0-7.6-2-15-5.8-21.6L361.6 5.5C359.6 2.1 356 0 352 0s-7.6 2.1-9.6 5.5L317.8 47.6zM128 176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48c-35.3 0-64 28.7-64 64v71c8.3 5.2 18.1 9 28.8 9c13.5 0 27.2-6.1 38.4-13.4c5.4-3.5 9.9-7.1 13-9.7c1.5-1.3 2.7-2.4 3.5-3.1c.4-.4 .7-.6 .8-.8l.1-.1 0 0 0 0s0 0 0 0s0 0 0 0c3.1-3.2 7.4-4.9 11.9-4.8s8.6 2.1 11.6 5.4l0 0 0 0 .1 .1c.1 .1 .4 .4 .7 .7c.7 .7 1.7 1.7 3.1 3c2.8 2.6 6.8 6.1 11.8 9.5c10.2 7.1 23 13.1 36.3 13.1s26.1-6 36.3-13.1c5-3.5 9-6.9 11.8-9.5c1.4-1.3 2.4-2.3 3.1-3c.3-.3 .6-.6 .7-.7l.1-.1c3-3.5 7.4-5.4 12-5.4s9 2 12 5.4l.1 .1c.1 .1 .4 .4 .7 .7c.7 .7 1.7 1.7 3.1 3c2.8 2.6 6.8 6.1 11.8 9.5c10.2 7.1 23 13.1 36.3 13.1s26.1-6 36.3-13.1c5-3.5 9-6.9 11.8-9.5c1.4-1.3 2.4-2.3 3.1-3c.3-.3 .6-.6 .7-.7l.1-.1c2.9-3.4 7.1-5.3 11.6-5.4s8.7 1.6 11.9 4.8l0 0 0 0 0 0 .1 .1c.2 .2 .4 .4 .8 .8c.8 .7 1.9 1.8 3.5 3.1c3.1 2.6 7.5 6.2 13 9.7c11.2 7.3 24.9 13.4 38.4 13.4c10.7 0 20.5-3.9 28.8-9V288c0-35.3-28.7-64-64-64V176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48H256V176c0-17.7-14.3-32-32-32s-32 14.3-32 32v48H128V176zM448 394.6c-8.5 3.3-18.2 5.4-28.8 5.4c-22.5 0-42.4-9.9-55.8-18.6c-4.1-2.7-7.8-5.4-10.9-7.8c-2.8 2.4-6.1 5-9.8 7.5C329.8 390 310.6 400 288 400s-41.8-10-54.6-18.9c-3.5-2.4-6.7-4.9-9.4-7.2c-2.7 2.3-5.9 4.7-9.4 7.2C201.8 390 182.6 400 160 400s-41.8-10-54.6-18.9c-3.7-2.6-7-5.2-9.8-7.5c-3.1 2.4-6.8 5.1-10.9 7.8C71.2 390.1 51.3 400 28.8 400c-10.6 0-20.3-2.2-28.8-5.4V480c0 17.7 14.3 32 32 32H416c17.7 0 32-14.3 32-32V394.6z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Bake
        </button>
      </div>
    </div>
    <div
      className="d-flex flex-column background-gray overflow-none-100 col"
    >
      <div
        className="d-flex flex-column background-white h-50"
      >
        <div
          className="title-block"
        >
          <h3>
            Obfuscated
          </h3>
          <div
            className="d-flex flex-row"
          >
            <div
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-trash "
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="fill-under-title"
        >
          <div
            className="position-relative background-white flex-grow-1 mh-100 h-100"
            data-testid="obfuscated-editor"
          />
        </div>
      </div>
      <div
        className="d-flex flex-column background-white h-50 mh-50"
      >
        <div
          className="title-block"
        >
          <h3>
            Deobfuscated 
          </h3>
          <div
            className="d-flex flex-row"
          >
            <div
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-copy "
                data-icon="copy"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M272 0H396.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H272c-26.5 0-48-21.5-48-48V48c0-26.5 21.5-48 48-48zM48 128H192v64H64V448H256V416h64v48c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="fill-under-title"
        >
          <div
            className="h-100 align-items-center justify-content-center d-none"
          >
            <div
              className="spinner-border"
              role="status"
            />
          </div>
          <div
            className="position-relative background-white flex-grow-1 mh-100 h-100 "
            data-testid="deobfuscated-editor"
            hidden={false}
          />
        </div>
      </div>
    </div>
    <div
      className="d-flex flex-column background-gray overflow-none-100 col"
    >
      <div
        className="d-flex flex-column background-white h-50"
      >
        <div
          className="title-block"
          style={
            {
              "paddingRight": 75,
            }
          }
        >
          <h3>
            Sandbox
          </h3>
          <div
            className="d-flex flex-row"
          >
            <div
              aria-label="run"
              data-testid="run"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-play "
                data-icon="play"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </div>
            <div
              aria-label="clear"
              data-testid="clear"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-trash "
                data-icon="trash"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="fill-under-title"
        >
          <div
            className="background-white mb-3 p-2 overflow-auto h-100 mh-100"
            onKeyDown={[Function]}
          />
        </div>
      </div>
      <div
        className="d-flex flex-column background-white h-50 mh-50"
      >
        <div
          className="title-block"
        >
          <h3>
            Output
          </h3>
          <div
            className="d-flex flex-row"
          />
        </div>
        <div
          className="fill-under-title"
        >
          <div
            className="output background-white mb-3 p-2 overflow-auto overflow-none-100"
          >
            <div
              className="d-flex align-items-center mt-3"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-chevron-left "
                data-icon="chevron-left"
                data-prefix="fas"
                focusable="false"
                opacity={0.2}
                role="img"
                style={{}}
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
              <pre
                className="ms-2 mb-0"
              >
                undefined
              </pre>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
